### PR TITLE
Conv BN folding data type issue when conv has no bias

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
PR https://github.com/pytorch/pytorch/pull/77042 has fixed the new folding conv-bn data type issue but missing the case when original conv has no bias input.
In this PR:

- Fix the new folding conv-bn's bias data type issue, when conv has no bias but weight as lower precision datatype, the new generated bias data type should be same as conv's weight.  
- Move the Autocast JIT Trace UT from `test_jit.py` to `test_jit_autocast.py`.

### CODE REVIEW

File: torch/csrc/jit/passes/frozen_conv_folding.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ bool FoldFrozenConvBatchnorm(Block* b) {
         at::ScalarType bias_dtype = bn_rm.scalar_type();
         at::ScalarType weight_dtype = conv_w.scalar_type();
         at::DeviceType weight_device = conv_w.device().type();
-        if (weight_device == at::kCUDA &&
+        if ((weight_device == at::kCUDA || weight_device == at::kCPU) &&
```
Comment (davidberard98) [inline torch/csrc/jit/passes/frozen_conv_folding.cpp:91]: instead could we just remove the device type check here?

**Review by davidberard98 — APPROVED** [review]
LGMT - but can you address the comment I left?

File: torch/csrc/jit/passes/frozen_conv_folding.cpp (hunk 1 above)
Comment (leslie-fang-intel) [reply torch/csrc/jit/passes/frozen_conv_folding.cpp]: @davidberard98 Thanks for the comment. Remove the device type check. Please help to take a look again.

**Comment by davidberard98** [comment]
@leslie-fang-intel looks good to me - let's wait until the tests finish just in case, and then we can merge it

**Pull request closed without merging**